### PR TITLE
compilation-mode: maps `gf` to `find-file-at-point`

### PR DESCRIPTION
**Pull request by FrancescElies**
Hi,

This PR maybe solves https://github.com/syl20bnr/spacemacs/issues/7702

I found two ways of adding the keybinding, but not sure if this is the right thing and the right place.

```
;; option 1
  (define-key evil-motion-state-map "gf" 'find-file-at-point)
;; option 2
  (evil-define-key 'motion compilation-mode-map (kbd "gf") 'find-file-at-point)
```

Guidance would be highly appreciated.

All best

### CODE REVIEW

**Comment by bmag** [comment]
`(kbd "gf")` returns `"gf"`, so there isn't any difference. In other cases there is a difference, for example to bind the return key you'd use `(kbd "RET")` but not `"RET"`.

**Comment by bmag** [comment]
Also, we probably want to bind the keys in [spacemacs-base](https://github.com/syl20bnr/spacemacs/blob/58c8fd515a9a0ab9bde21cfb531a5541c353b77a/layers/+distributions/spacemacs-base/keybindings.el#L236), not spacemacs-bootstrap.

File: layers/+distributions/spacemacs-base/keybindings.el [hunk 1]
```diff
@@ -235,6 +235,7 @@
   "cd" 'spacemacs/close-compilation-window)
 (with-eval-after-load 'compile
   (define-key compilation-mode-map "r" 'recompile)
+  (define-key evil-motion-state-map "gf" 'find-file-at-point)
   (define-key compilation-mode-map "g" nil))
```
Comment (FrancescElies) [inline layers/+distributions/spacemacs-base/keybindings.el]: Just being curious, what is `(define-key compilation-mode-map "g" nil)` doing, is this really needed?

**Comment by FrancescElies** [comment]
keybinding moved to spacemacs-base

File: layers/+distributions/spacemacs-base/keybindings.el [hunk 2]
```diff
@@ -235,6 +235,7 @@
   "cd" 'spacemacs/close-compilation-window)
 (with-eval-after-load 'compile
   (define-key compilation-mode-map "r" 'recompile)
+  (define-key evil-motion-state-map "gf" 'find-file-at-point)
```
Comment (bmag) [inline layers/+distributions/spacemacs-base/keybindings.el]: Should use evil-define-key like before

Comment (FrancescElies) [inline layers/+distributions/spacemacs-base/keybindings.el]: Updated https://github.com/syl20bnr/spacemacs/pull/7712/commits/12517cf695043b49ad7cc3c489af37cb0c270ab6
Is this one fine?

Comment (bmag) [inline layers/+distributions/spacemacs-base/keybindings.el]: Yes (but I didn't test it locally yet)

Comment (FrancescElies) [inline layers/+distributions/spacemacs-base/keybindings.el]: I still could not understand the difference between both if any, if I remember correctly both did the same. 

Thanks for your help 👍

**Comment by bmag** [comment]
Thank you :+1: Cherry-picked into develop, you can safely delete your branch.

**Pull request closed without merging**